### DISCUSSION TITLE
The type initializer for 'PerTypeValues`1' threw an exception

### DISCUSSION THREAD

**Discussion by hamelpierreluc** [empira/PDFsharp#220] [Q&A]
Hello, I have an error when opening a PDF, which seems normal to me. The error appears to indicate that there is a mismatch in the DLL version. It mentions version 4.0.4.1, whereas I have version 6.0 of this DLL, which comes automatically with the 6.2-preview package. I have no idea what I can do about it.

In the stack trace, it shows that the error occurs when logging a key value, but I have no idea which key it is referring to.

Thank you for taking the time to help me!

Here is the error:

> The type initializer for 'PerTypeValues`1' threw an exception.
> Message de l'exception d'origine
> --------------------------------
> Could not load file or assembly 'System.Runtime.CompilerServices.Unsafe, Version=4.0.4.1, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)

Here is the stackTrace:

>    at Microsoft.Extensions.Logging.LogValuesFormatter..ctor(String format)
>    at Microsoft.Extensions.Logging.FormattedLogValues.<>c.<.ctor>b__9_0(String f)
>    at System.Collections.Concurrent.ConcurrentDictionary`2.GetOrAdd(TKey key, Func`2 valueFactory)
>    at Microsoft.Extensions.Logging.FormattedLogValues..ctor(String format, Object[] values)
>    at Microsoft.Extensions.Logging.LoggerExtensions.Log(ILogger logger, LogLevel logLevel, EventId eventId, Exception exception, String message, Object[] args)
>    at PdfSharp.Pdf.IO.Parser.TryReadEndStream(PdfDictionary dict, Int64 streamStart, Int32& streamLength, SuppressExceptions suppressObjectOrderExceptions)
>    at PdfSharp.Pdf.IO.Parser.ReadDictionaryStream(PdfDictionary dict, SuppressExceptions suppressObjectOrderExceptions)
>    at PdfSharp.Pdf.IO.Parser.ReadObjectInternal(PdfObject pdfObject, PdfObjectID objectID, Boolean includeReferences, Boolean fromObjectStream, SuppressExceptions suppressObjectOrderExceptions)
>    at PdfSharp.Pdf.IO.Parser.ReadIndirectObject(PdfReference pdfReference, SuppressExceptions suppressObjectOrderExceptions, Boolean withoutDecrypting)
>    at PdfSharp.Pdf.IO.Parser.ReadAllIndirectObjects()
>    at PdfSharp.Pdf.IO.PdfReader.OpenFromStream(Stream stream, String password, PdfDocumentOpenMode openMode, PdfPasswordProvider passwordProvider, PdfReaderOptions options)
>    at

**Comment by ThomasHoevel** [comment]
You probably get compile-time warnings about the version conflict.

We cannot reproduce your problem without a complete project.

If you download the PDFsharp source code, you can compile PDFsharp with different package requirements. Thus you should be able to resolve that issue.
